### PR TITLE
Gdxdsd 3286

### PR DESCRIPTION
**Pull request by BJenning**

### CODE REVIEW

File: microservices/s3_to_redshift/asset_data_to_redshift.py [hunk 1]
```diff
@@ -56,15 +57,22 @@
 handler.setFormatter(formatter)
 logger.addHandler(handler)
 
+
+def clean_exit(code, message):
+    """Exits with a logger message and code"""
+    logger.info('Exiting with code %s : %s', str(code), message)
```
Comment (mark-walle) [inline microservices/s3_to_redshift/asset_data_to_redshift.py:63]: @BJenning  as @danpollock  mentioned, he would like these posted to debug. Please update.

File: microservices/s3_to_redshift/asset_data_to_redshift.py [hunk 2]
```diff
@@ -316,26 +339,46 @@ def is_processed(object_summary):
...
+            outfile = badfile
         else:
             logger.warning('File not empty, keying to badfile \
                            and proceeding.')
             outfile = badfile
-        client.copy_object(Bucket="sp-ca-bc-gov-131565110619-12-microservices",
-                           CopySource="sp-ca-bc-gov-\
-                           131565110619-12-microservices/"
-                           + object_summary.key, Key=outfile)
-        continue
```
Comment (mark-walle) [inline microservices/s3_to_redshift/asset_data_to_redshift.py:338]: @BJenning does it make sense to halt the microservice operation if the outfile = badfile here and posts an exit code of 1 instead?

File: microservices/s3_to_redshift/asset_data_to_redshift.py [hunk 3]
```diff
@@ -360,8 +403,8 @@ def is_processed(object_summary):
     if 'dateformat' in data:
         for thisfield in data['dateformat']:
             df[thisfield['field']] = \
-                pd.to_datetime(df[thisfield['field']],
-                               format=thisfield['format'])
+                pandas.to_datetime(df[thisfield['field']],
```
Comment (mark-walle) [inline microservices/s3_to_redshift/asset_data_to_redshift.py:406]: Why not pd?
you `import pandas as pd` at the top.

**Review by mark-walle — CHANGES_REQUESTED** [review]

File: microservices/s3_to_redshift/asset_data_to_redshift.py (hunk 2 above)
Comment (BJenning) [reply microservices/s3_to_redshift/asset_data_to_redshift.py:338]: Isn't that what lines 362 to 369 are doing? Copying the file to the bad folder, and then calling clean_exit...that should handle the pandas error?

**Review by mark-walle — APPROVED** [review]

File: microservices/s3_to_redshift/asset_data_to_redshift.py (hunk 2 above)
Comment (mark-walle) [reply microservices/s3_to_redshift/asset_data_to_redshift.py:338]: > Isn't that what lines 362 to 369 are doing? Copying the file to the bad folder, and then calling clean_exit...that should handle the pandas error?

You're correct, I misread the commit record when I made this comment.

**Pull request merged**